libcontainer/nsenter/nsexec.c

```diff
@@ -741,6 +754,22 @@ void nsexec(void)
 					bail("failed to dup stderr");
 			}
 
+			/* ... wait until our topmost parent has finished cgroup setup in p.manager.Apply() ... */
+			if (config.cloneflags & CLONE_NEWCGROUP) {
+				len = read(pipenum, buf, SYNC_BUF_LEN);
+				if (len < 0)
+					bail("read synchronisation value failed");
+				value = strtol(buf, &endptr, 10);
```

Comment:
Yeah, this shouldn't be necessary if you just write a single byte. Also, why is value not local to this block?